restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/mybatis/MybatisApplicationsDao.java

```diff
@@ -94,6 +95,39 @@ private Application getApplication(final String selector, final String parameter
     }
 
     @Override
+    public List<Application> getApplications(Sid accountSid, boolean includeNumbers) {
+        if (!includeNumbers)
+            return getApplications(accountSid);
+
+        final SqlSession session = sessions.openSession();
+        try {
+            final List<Map<String, Object>> results = session.selectList(namespace + "getApplicationsAndNumbers", accountSid.toString());
+            final List<Application> applications = new ArrayList<Application>();
+            if (results != null && !results.isEmpty()) {
+                Application previousApp = null;
+                Application app = null;
+                for (final Map<String, Object> result : results) {
+                    app = toApplication(result);
+                    if (previousApp != null && previousApp.getSid().equals(app.getSid()) )
```

Comment:
is this based on query results being order by appId, i didnt see any orderBy clause when the query was specified?